lib/cache.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+const reportErrorRedis = function( error ) {
+    if ( error !== null ) {
+        console.log( error );
+    }
+};
+const Redis         = require( 'redis' ).createClient().on( 'error', reportErrorRedis );
+const cacheInMemory = {};
+const REQUEST_OPTIONS = {};
+
+Redis.keys('cache:userAgentParser:*', function( error, userAgentKeyList ) {
+    if ( error !== null ) {
+        return reportErrorRedis({ from:'Load userAgentKeyList', error: error });
+    }
+
+    for ( let i = 0, j = userAgentKeyList.length; i < j; i++ ) {
```

Comment:
can improve performance by setting length above the for loop statement. as of now the .length line would run every iteration, while if its constant it only needs to run once.